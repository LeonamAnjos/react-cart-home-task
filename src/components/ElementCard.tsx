import React, { ReactNode } from "react";
import Card from "@mui/material/Card";
import CardActions from "@mui/material/CardActions";
import CardHeader from "@mui/material/CardHeader";
import CardContent from "@mui/material/CardContent";

type ElementCardProps = {
  title: string;
  subTitle?: string;
  children?: ReactNode;
  avatar?: ReactNode;
  actions?: ReactNode;
};

const ElementCard = (props: ElementCardProps) => {
  const { title, subTitle, avatar, children, actions } = props;

  return (
    <Card sx={{ m: 2 }}>
      <CardHeader avatar={avatar} title={title} subheader={subTitle} />
      <CardContent>{children}</CardContent>
      <CardActions disableSpacing>{actions}</CardActions>
    </Card>
  );
};

export default ElementCard;
