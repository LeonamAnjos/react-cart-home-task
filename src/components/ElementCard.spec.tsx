import { render } from "@testing-library/react";
import ElementCard from "./ElementCard";

describe("ElementCard", () => {
  it("WHEN rendered", () => {
    const { container } = render(
      <ElementCard
        title={"Title test"}
        avatar={"Avatar test"}
        subTitle={"SubTitle test"}
        actions={"Actions test"}
      >
        {"Content test"}
      </ElementCard>
    );

    expect(container).toMatchSnapshot();
  });
});
