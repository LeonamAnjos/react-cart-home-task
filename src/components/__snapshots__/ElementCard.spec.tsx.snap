// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementCard WHEN rendered 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-16cqd0p-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-avatar css-1ssile9-MuiCardHeader-avatar"
      >
        Avatar test
      </div>
      <div
        class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-title css-et1ao3-MuiTypography-root"
        >
          Title test
        </span>
        <span
          class="MuiTypography-root MuiTypography-body2 MuiCardHeader-subheader css-83ijpv-MuiTypography-root"
        >
          SubTitle test
        </span>
      </div>
    </div>
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      Content test
    </div>
    <div
      class="MuiCardActions-root css-1rwjz6-MuiCardActions-root"
    >
      Actions test
    </div>
  </div>
</div>
`;
